ucp/_libs/core.pyx

```diff
@@ -252,19 +252,20 @@ async def listener_handler(ucp_endpoint, ctx, ucp_worker, func, guarantee_msg_or
         await _func(pub_ep)
 
 
-cdef void _listener_callback(ucp_ep_h ep, void *args):
+cdef void _listener_callback(ucp_ep_h ep, void *args) nogil:
     cdef _listener_callback_args *a = <_listener_callback_args *> args
-    cdef object ctx = <object> a.py_ctx
-    cdef object func = <object> a.py_func
-    asyncio.ensure_future(
-        listener_handler(
-            int(<uintptr_t><void*>ep),
-            ctx,
-            int(<uintptr_t><void*>a.ucp_worker),
-            func,
-            a.guarantee_msg_order
+    with gil:
```

Comment:
This needs to be with the gil ?

Reply:
Oh , any call back needs to be with the gil, correct ?

Reply:
That's right.